dspace/modules/payment-system/payment-api/src/main/java/org/dspace/paymentsystem/PaymentServiceImpl.java

```diff
@@ -107,12 +110,12 @@ public String generateSecureToken(ShoppingCart shoppingCart, String secureTokenI
             }
             String amount = "0.00";
 
-            if (type.equals("S")) {
+            if (transactionType.equals(PAYPAL_SALE)) {
                 //generate reauthorization form
                 amount = Double.toString(shoppingCart.getTotal());
             }
 
-            String urlParameters = "SECURETOKENID=" + secureTokenId + "&CREATESECURETOKEN=Y" + "&MODE=" + ConfigurationManager.getProperty("payment-system", "paypal.mode") + "&PARTNER=" + ConfigurationManager.getProperty("payment-system", "paypal.partner") + "&VENDOR=" + ConfigurationManager.getProperty("payment-system", "paypal.vendor") + "&USER=" + ConfigurationManager.getProperty("payment-system", "paypal.user") + "&PWD=" + ConfigurationManager.getProperty("payment-system", "paypal.pwd") + "&TENDER=" + type + "&TRXTYPE=" + type + "&FIRSTNAME=" + userFirstName + "&LASTNAME=" + userLastName + "&COMMENT1=" + userName + "&COMMENT2=" + userEmail + "&AMT=" + amount + "&CURRENCY=" + shoppingCart.getCurrency();
+            String urlParameters = "SECURETOKENID=" + secureTokenId + "&CREATESECURETOKEN=Y" + "&MODE=" + ConfigurationManager.getProperty("payment-system", "paypal.mode") + "&PARTNER=" + ConfigurationManager.getProperty("payment-system", "paypal.partner") + "&VENDOR=" + ConfigurationManager.getProperty("payment-system", "paypal.vendor") + "&USER=" + ConfigurationManager.getProperty("payment-system", "paypal.user") + "&PWD=" + ConfigurationManager.getProperty("payment-system", "paypal.pwd") + "&TENDER=" + transactionType + "&TRXTYPE=" + transactionType + "&FIRSTNAME=" + userFirstName + "&LASTNAME=" + userLastName + "&COMMENT1=" + userName + "&COMMENT2=" + userEmail + "&AMT=" + amount + "&CURRENCY=" + shoppingCart.getCurrency();
```

Comment:
Possibly these extra parameters are just ignored when generating the token, because the actual transaction isn't happening. Just below here, TENDER is correctly set to "C".